test/integration/prerender/test/index.test.js

```diff
@@ -887,48 +887,44 @@ const runTests = (dev = false, looseMode = false) => {
           dataRouteRegex: normalizeRegEx(
             `^\\/_next\\/data\\/${escapedBuildId}\\/blog\\/([^\\/]+?)\\.json$`
           ),
-          routeRegex: normalizeRegEx('^\\/blog\\/([^\\/]+?)(?:\\/)?$'),
+          routeRegex: normalizeRegEx('^\\/blog\\/([^\\/]+?)$'),
         },
         '/blog/[post]/[comment]': {
           fallback: '/blog/[post]/[comment].html',
           dataRoute: `/_next/data/${buildId}/blog/[post]/[comment].json`,
           dataRouteRegex: normalizeRegEx(
             `^\\/_next\\/data\\/${escapedBuildId}\\/blog\\/([^\\/]+?)\\/([^\\/]+?)\\.json$`
           ),
-          routeRegex: normalizeRegEx(
-            '^\\/blog\\/([^\\/]+?)\\/([^\\/]+?)(?:\\/)?$'
-          ),
+          routeRegex: normalizeRegEx('^\\/blog\\/([^\\/]+?)\\/([^\\/]+?)$'),
         },
         '/non-json/[p]': {
           dataRoute: `/_next/data/${buildId}/non-json/[p].json`,
           dataRouteRegex: normalizeRegEx(
             `^\\/_next\\/data\\/${escapedBuildId}\\/non\\-json\\/([^\\/]+?)\\.json$`
           ),
           fallback: '/non-json/[p].html',
-          routeRegex: normalizeRegEx('^\\/non\\-json\\/([^\\/]+?)(?:\\/)?$'),
+          routeRegex: normalizeRegEx('^\\/non\\-json\\/([^\\/]+?)$'),
         },
         '/user/[user]/profile': {
           fallback: '/user/[user]/profile.html',
           dataRoute: `/_next/data/${buildId}/user/[user]/profile.json`,
           dataRouteRegex: normalizeRegEx(
             `^\\/_next\\/data\\/${escapedBuildId}\\/user\\/([^\\/]+?)\\/profile\\.json$`
           ),
-          routeRegex: normalizeRegEx(
-            `^\\/user\\/([^\\/]+?)\\/profile(?:\\/)?$`
-          ),
+          routeRegex: normalizeRegEx(`^\\/user\\/([^\\/]+?)\\/profile$`),
         },
         '/catchall/[...slug]': {
           fallback: '/catchall/[...slug].html',
           routeRegex: normalizeRegEx('^\\/catchall\\/(.+?)(?:\\/)?$'),
           dataRoute: `/_next/data/${buildId}/catchall/[...slug].json`,
           dataRouteRegex: normalizeRegEx(
-            `^\\/_next\\/data\\/${escapedBuildId}\\/catchall\\/(.+?)\\.json$`
+            `^\\/_next\\/data\\/${escapedBuildId}\\/catchall\\/(.+?)(?:\\/)?\\.json$`
```

Comment:
This regex does not look correct. It allows for a trailing slash right before `.json`?

Compiles to:
```
/^\/_next\/data\/abc\/catchall\/(.+?)(?:\/)?\.json$/
```

Invalidly matches:

```
/_next/data/abc/catchall/abc/.json
```

![image](https://user-images.githubusercontent.com/616428/83154198-d590d500-a0cd-11ea-84a8-54369602402d.png)

Reply:
Right, good catch. Updated